vaadin-button.html

```diff
@@ -22,25 +22,186 @@
+      }
+
+      :host(:hover) {
+        border-color: #666;
+      }
+
+      :host([disabled]) {
+        color: #aaa !important;
+        background: #eee !important;
+        border-color: #aaa !important;
+        pointer-events: none;
+      }
+
+      :host([active]) {
+        background: linear-gradient(to bottom right, #eee, #fff);
       }
     </style>
+    <slot></slot>
   </template>
 
   <script>
-    class VaadinElement extends Polymer.Element {
+    if (!window.Vaadin) window.Vaadin = {};
+
+    const ENTER = 13;
+    const SPACE = 32;
+
+    Vaadin.ButtonElement = class VaadinButtonElement extends Polymer.Element {
       static get is() {
         return 'vaadin-button';
       }
 
       static get config() {
         return {
           properties: {
+            /**
+             * If true, the button is in the active state (clicked).
+             */
+            active: {
+              type: Boolean,
+              default: false,
+              reflectToAttribute: true
+            },
+
+            /**
+             * Specify that this control should have input focus when the page loads.
+             */
+            autofocus: {
+              type: Boolean
+            },
+
+            /**
+             * Indicates that this control is not available for interaction.
+             */
+            disabled: {
+              type: Boolean,
+              default: false,
+              reflectToAttribute: true,
+              observer: '_disabledChanged',
+            },
+
+            /**
+             * role="button"
+             */
+            role: {
+              type: String,
+              value: 'button',
+              reflectToAttribute: true,
+              readOnly: true
+            },
+
+            /**
+             * The position of the element in the tabbing navigation order for the current document.
+             */
+            tabindex: {
+              type: Number,
+              value: 0,
+              reflectToAttribute: true
+            },
+
+            /**
+             *
+             */
+            _previousTabIndex: Number
           }
         };
       }
-    }
 
-    customElements.define(VaadinElement.is, VaadinElement);
+      ready() {
+        this._addListeners();
+
+        super.ready();
+
+        if (this.autofocus) {
+          window.requestAnimationFrame(function() {
+            this.focus();
+          }.bind(this));
+        }
+      }
+
+      activate(e) {
+        this.active = true;
+      }
+
+      deactivate(e) {
+        this.active = false;
+      }
+
+      _disabledChanged(disabled) {
+        if (disabled) {
+          this._previousTabIndex = this.tabindex;
+          this.tabindex = null;
+          this._removeListeners();
+        } else {
+          this.tabindex = this._previousTabIndex;
+          this._addListeners();
+        }
+      }
+
+      _addListeners() {
+        this.addEventListener('down', this._handleDown.bind(this));
```

Comment:
Done.

For some reason `bind(this)` is needed for Shadow DOM, otherwise tests are failing.